test/mongoConnectionStrings.test.ts

```diff
@@ -62,15 +62,14 @@ suite(`mongoCollectionStrings`, () => {
         testDatabaseNameFromConectionString(`mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/abc`, 'abc');
 
         // special characters
-        testDatabaseNameFromConectionString(`mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/abc.def`, 'abc.def');
-        testDatabaseNameFromConectionString(`mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/abc.def-ghi_jkl`, 'abc.def-ghi_jkl');
```

Comment:
`.` is not an allowed character for mongo servers. 
https://docs.mongodb.com/manual/reference/limits/#naming-restrictions